private/buf/cmd/buf/command/decode/decode.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package decode
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufanalysis"
+	"github.com/bufbuild/buf/private/bufpkg/bufreflect"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/protoencoding"
+	"github.com/bufbuild/buf/private/pkg/stringutil"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	errorFormatFlagName = "error-format"
+	sourceFlagName      = "source"
+	typeFlagName        = "type"
+	outputFlagName      = "output"
+	outputFlagShortName = "o"
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appflag.Builder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <descriptor>",
+		Short: "Decode binary descriptors with a source reference.",
+		Long: `The first argument is the serialized descriptor to decode.
+If no argument is specified, defaults to stdin.`,
```

Comment:
We also need to update this now that we require the message on stdin.

We should rephrase it to refer to a serialized _message_ instead of a descriptor (my fault on that earlier).